R/reactives.R

```diff
@@ -1728,6 +1729,18 @@ reactivePoll <- function(intervalMillis, session, checkFunc, valueFunc) {
 
   }, label = NULL)
 
+  # When no one holds a reference to this object anymore, destroy and remove the
+  # observer so that it doesn't keep firing, and hold onto resources.
+  reg.finalizer(attr(re, "observable"), function(e) {
+    message("finalizing reactive")
+    o$destroy()
```

Comment:
I'm scared of this... can we do (a lot) less work inside the finalizer, preferably all in C?

How about we make our own version of reg.finalizer, that's built on top of reg.finalizer but instead of invoking your callback inline with reg.finalizer, it just schedules your callback to be invoked at a more convenient moment? (The callback would need to not access the object being finalized)

Reply:
I don't think we need to use C to schedule things to happen later. (If we do use C, that would be new for Shiny.) When do you have in mind for running the actual finalizers?

Reply:
Would `later(callback, 0)` be a reasonable solution?

Reply:
Note: the original `reg.finalizer` call has been replaced with `safe_finalizer` here.